tests/unit/test_client.py

```diff
@@ -104,6 +107,23 @@ def to_bytes_iterator(content, chunk_size=64 * 1024, exception_to_raise=None):
             raise exception_to_raise
 
 
+def mock_response(expected_range, blob_length, content, exception=None, etag=blob_etag):
+    response = mock.Mock(spec=StreamDownloadGenerator)
+    response.response = mock.Mock()
+    response.response.headers = {
+        "Content-Range": f"bytes {expected_range}/{blob_length}",
+        "ETag": etag,
+    }
+    response.iter_content_func = iter(
+        to_bytes_iterator(
+            content=slice_bytes(content, expected_range), exception_to_raise=exception
+        )
+    )
+    response.__iter__ = mock.Mock()
```

Comment:
Instead of setting a `Mock()` object like this, we can just make the `response` a [`MagicMock`](https://docs.python.org/3/library/unittest.mock.html#magicmock-and-magic-method-support) which will allow us to mock the `__iter__` return value.